public/video-ui/src/components/Iconik/IconikProjectPicker.tsx

```diff
@@ -0,0 +1,252 @@
+import { uniqueId } from 'lodash';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getVideo } from '../../actions/VideoActions/getVideo';
+import { saveVideo } from '../../actions/VideoActions/saveVideo';
+import { IconikData, Video } from '../../services/VideosApi';
+import {
+  fetchIconikCommissions,
+  fetchIconikProjects,
+  IconikState,
+  resetCommissions,
+  resetProjects
+} from '../../slices/iconik';
+import { AppDispatch, RootState } from '../../util/setupStore';
+
+type Props = {
+  video: Video;
+};
+
+export const IconikProjectPicker = ({ video }: Props) => {
+  const dispatch = useDispatch<AppDispatch>();
+
+  const { workingGroups, commissions, projects } = useSelector<
+    RootState,
+    IconikState
+  >(({ iconik }) => iconik);
+
+  const [workingGroup, setWorkingGroup] = React.useState<string | undefined>(
+    () => video.iconikData?.workingGroupId
+  );
+  const [commission, setCommission] = React.useState<string | undefined>(
+    () => video.iconikData?.commissionId
+  );
+  const [project, setProject] = React.useState<string | undefined>(
+    () => video.iconikData?.projectId
+  );
+
+  useEffect(() => {
+    if (workingGroup) {
+      dispatch(fetchIconikCommissions(workingGroup));
+    } else {
+      dispatch(resetCommissions());
+    }
+    if (commission) {
+      dispatch(fetchIconikProjects(commission));
+    } else {
+      dispatch(resetProjects());
+    }
+  }, [commission, dispatch, workingGroup]);
+
+  const hasBeenEdited =
```

Comment:
Is this just computed once on load? Or should this be wrapped in a callback and have the form fields as state values?